plugins/modules/inventory_intent.py

```diff
@@ -961,10 +1089,162 @@ def get_diff_merged(self, config):
+                response = response.get("response")
+                task_id = response.get("taskId")
+                # With this task ID call the Get Task Details API
+                task_resp = self.dnac._exec(
+                    family="task",
+                    function='get_task_by_id',
+                    op_modifies=True,
+                    params={"task_id": task_id},
+                )
+
+                while True:
+                    execution_details = self.get_task_details(task_id)
+
+                    if execution_details.get("additionalStatusURL"):
+                        file_id = execution_details.get("additionalStatusURL").split("/")[-1]
+                        break
+                    elif execution_details.get("isError") and execution_details.get("failureReason"):
+                        failed_reason = execution_details.get("failureReason")
+                        msg = "Could not get the File ID because of {0} so can't export device details in csv file".format(failed_reason)
+                        self.msg = msg
+                        self.log(msg)
+                        self.status = "failed"
+                        return self
+
+                # With this File ID call the Download File by FileID API
+                response = self.dnac._exec(
+                    family="file",
+                    function='download_a_file_by_fileid',
+                    op_modifies=True,
+                    params={"file_id": file_id},
+                )
+
+                device_data = []
+                encoded_resp = response.data.decode(encoding='utf-8')
+                self.log(str(encoded_resp))
+
+                # Parse the CSV-like string into a list of dictionaries
+                csv_reader = csv.DictReader(StringIO(encoded_resp))
+                for row in csv_reader:
+                    device_data.append(row)
+
+                current_date = datetime.now()
+                formatted_date = current_date.strftime("%m-%d-%Y")
+                file_name = "devices-" + str(formatted_date) + ".csv"
+
+                # Write the data to a CSV file
+                with open(file_name, 'w', newline='') as csv_file:
+                    fieldnames = device_data[0].keys()
+                    csv_writer = csv.DictWriter(csv_file, fieldnames=fieldnames)
+                    csv_writer.writeheader()
+                    csv_writer.writerows(device_data)
+
+                msg = "Device Details Exported Successfully to the CSV file - {0}".format(file_name)
+                self.log(msg)
+                self.status = "success"
+                self.result['changed'] = True
+                self.result['log'].append(msg)
+
+            except Exception as e:
+                msg = "Cannot Export the Device Details into CSV file for {0}".format(str(device_ips))
+                self.log(msg)
+                self.status = "failed"
+                self.msg = msg
+
```

Comment:
Don't we need to close the file before returning?

Reply:
File is automatically closed when you exit the 'with' block